articles/coding-on-purpose.md

```diff
@@ -0,0 +1,99 @@
+
+No matter whether all this talk is true, one thing is clear: making an impact is an economical necessity if your business wants to survive in the long run. Or as Paul Graham of *Y Combinator* - note the Y in their name - says:
+
+> Make something people want.
+
+By the way: Justin Bieber - born 1994 - is a millenial too. And so it's no surprise that his current (2017) world tour is titled the *Purpose World Tour*.
+
+When building software, the *why* or better the *what for* question is a very important one. I'm aware that this is nothing new for you. But this is what makes it easy to forget and to lose the focus on it. We usually start our projects with a more or less clear vision of what we want to achieve - how the world will be better when we're done. Then we start building. We start setting milestones, and soon reaching the milestone becomes our highest goal. In order to meet milestones we need to deliver a certain amount of user stories per sprint (or so we think...) and soon estimating effort in story points and tracking velocity become our way of steering the project. Or we might be busy optimizing the hourly rate of our sub-contractors from another continent. Whenever our focus drifts away from achieving the original purpose, we need to remind ourselves:
+
+> When we build software, our success is measured by whether the software can make the impact for which the customer is ready paying a certain amount of money.
+
+You don't need to be a millenial to understand this. This was true long before millenials were even born.
+
+## Purpose in Software Development Methodologies
+
+As long as there has been a professional software industry people in this industry have invented methods for approaching software development. There's one theme that shows up again and again in all methods: *managing purpose*.
+
+* In *RUP*, one tries to establish traceability from lower level items to higher level goals
+* If you work with *Use Cases* each Use Case represents a goal on a certain abstraction level and you can show how lower level Use Cases support higher level Use Cases in a use case diagram.
+* *User Stories* are usually formulated according to a template that also includes a *goal* and a *benefit* (e.t. "as a &lt;role&gt; I want to &lt;goal&gt; so that &lt;benefit&gt;").
+* With a *User Story Map* we can create a hierarchy of these *User Stories*.
+* An *Impact Map* traces down the high level goals to actors, impacts and finally deliverables
+* *Design Thinking* has a strong focus on *identifying and understanding customer needs*
+* The idea of the *Minimal Viable Product* makes the point that we should only build as much as is necessary for fulfilling the desired goal.
+

```

Comment:
I would consider the RUP example and the Use Case Diagram not so well suited examples here (which does not mean that this techniques are not useful, if done appropriately), but these techniques do not directly help or adddress the problem of not loosing focus on the purpose, at least less compared to the others. But maybe I not understand why you think this techniques help on that (ever seen something about the "Purpose" or "Impact" in a Use Case Diagram?).